hack/run-locally.sh

```diff
@@ -94,24 +121,186 @@ function wait_for_cluster() {
     fi
 }
 
+function print_usage() {
+    >&2 echo "Usage: $0 [mode] -c <cluster-dir> [options]
+
+$0 can be run in different modes:
+
+ override: override already rendered cluster manifests to run a local cluster-network-operator
+ attach:   attach to an existing cluster, stop cluster-network-operator, and restart with overrides
+ new:      install a new cluster using openshift-install and run the cluster-network-operator
+                  locally. This mode is assumed if the cluster-dir is not present or empty.
+
+The following option is required for all modes:
+
+ -c      the cluster installation temp directory; leave empty to create a cluster from scratch
+
+The following options are accepted for 'new' mode:
+
+ -f               the path to an openshift-install-created install-config.yaml file; if not given one will be created
+ -i               path to the openshift-install binary; if not given PATH will be searched
+ -m               custom network plugin container image name
+ -n [net plugin]  the name of the network plugin to deploy; one of [sdn|OpenShiftSDN|ovn|OVNKubernetes]
+
+The following environment variables are honored:
+ - CLUSTER_DIR: the cluster installation temp directory
+ - INSTALLER_PATH: the path to the openshift-install binary for 'new' mode
+ - INSTALL_CONFIG: the path to the install-config.yaml file for 'new' mode
+"
+}
+
+PLUGIN_IMAGE="${PLUGIN_IMAGE:-}"
+NETWORK_PLUGIN="OpenShiftSDN"
+IMAGE_ENV_KEY="NODE_IMAGE"
+CLUSTER_DIR="${CLUSTER_DIR:-}"
+INSTALLER_PATH="${INSTALLER_PATH:-}"
+INSTALL_CONFIG="${INSTALL_CONFIG:-}"
+MODE="${MODE:-${1:-""}}"
+
+case "${MODE}" in
+  override|attach)
+    OPTIND=2
+    while getopts "c:" opt; do
+        case $opt in
+            c) CLUSTER_DIR="${OPTARG}";;
+            \?)
+                echo "Invalid option: -${OPTARG}" >&2
+                exit 1
+                ;;
+            :)
+                echo "Option -${OPTARG} requires an argument." >&2
+                exit 1
+                ;;
+            *)
+                print_usage
+                exit 2
+                ;;
+        esac
+    done
+    ;;
+
+  new)
+    OPTIND=2
+    while getopts "c:f:i:m:n:" opt; do
+        case $opt in
+            c) CLUSTER_DIR="${OPTARG}";;
+            f) INSTALL_CONFIG="${OPTARG}";;
+            i) INSTALLER_PATH="${OPTARG}";;
+            m) PLUGIN_IMAGE="${OPTARG}";;
+            n)
+                case ${OPTARG} in
+                    ovn)
```

Comment:
This fallthrough is not working for me, I get "hack/run-locally.sh: line 193: syntax error near unexpected token `)'".
Doing something like "ovn" | "OVNKubernetes" works and would catch both.

Same for sdn/OpenShiftSDN

Reply:
@rcarrillocruz fixed, PTAL thanks!